Subject: [PATCH] GsmCdmaPhone: Fix GSM SIM card ICCID on NV sub CDMA devices

The commit 94ae0fe784f0701d3c3bb3c7575362dc6684300b
("GsmCdmaPhone: Return dummy ICCID serial for NV sub")
broke the reported ICCID for GSM SIM cards when used
in global capable NV subscription CDMA devices.

GSM SIM ICCID shouldn't depend on the selected CDMA
subscription source in any way.

diff --git a/src/java/com/android/internal/telephony/GsmCdmaPhone.java b/src/java/com/android/internal/telephony/GsmCdmaPhone.java
index c9501d4..928593c 100644
--- a/src/java/com/android/internal/telephony/GsmCdmaPhone.java
+++ b/src/java/com/android/internal/telephony/GsmCdmaPhone.java
@@ -741,11 +741,11 @@
 
     @Override
     public String getIccSerialNumber() {
-        if (mCdmaSubscriptionSource == CDMA_SUBSCRIPTION_NV) {
-            return DUMMY_NV_ICC_SERIAL;
-        }
         IccRecords r = mIccRecords.get();
         if (!isPhoneTypeGsm() && r == null) {
+            if (mCdmaSubscriptionSource == CDMA_SUBSCRIPTION_NV) {
+                return DUMMY_NV_ICC_SERIAL;
+            }
             // to get ICCID form SIMRecords because it is on MF.
             r = mUiccController.getIccRecords(mPhoneId, UiccController.APP_FAM_3GPP);
         }
@@ -754,11 +754,11 @@
 
     @Override
     public String getFullIccSerialNumber() {
-        if (mCdmaSubscriptionSource == CDMA_SUBSCRIPTION_NV) {
-            return DUMMY_NV_ICC_SERIAL;
-        }
         IccRecords r = mIccRecords.get();
         if (!isPhoneTypeGsm() && r == null) {
+            if (mCdmaSubscriptionSource == CDMA_SUBSCRIPTION_NV) {
+                return DUMMY_NV_ICC_SERIAL;
+            }
             // to get ICCID form SIMRecords because it is on MF.
             r = mUiccController.getIccRecords(mPhoneId, UiccController.APP_FAM_3GPP);
         }
